community/modules/compute/schedmd-slurm-gcp-v5-partition/variables.tf

```diff
@@ -0,0 +1,298 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+variable "slurm_cluster_name" {
+  type        = string
+  description = "Cluster name, used for resource naming and slurm accounting."
+
+  validation {
+    condition     = can(regex("(^[a-z][a-z0-9]*$)", var.slurm_cluster_name))
+    error_message = "Variable 'slurm_cluster_name' must be a match of regex '(^[a-z][a-z0-9]*$)'."
+  }
+}
+
+variable "project_id" {
+  description = "Project in which the HPC deployment will be created"
+  type        = string
+}
+
+variable "region" {
+  description = "The default region for Cloud resources"
+  type        = string
+}
+
+variable "partition_name" {
+  description = "The name of the slurm partition"
+  type        = string
+}
+
+variable "partition_conf" {
+  description = <<-EOD
+    Slurm partition configuration as a map.
+    See https://slurm.schedmd.com/slurm.conf.html#SECTION_PARTITION-CONFIGURATION
+    EOD
+  type        = map(string)
+  default     = {}
+}
+
+variable "is_default" {
+  description = <<-EOD
+    Sets this partition as the default partition by updating the partition_conf.
+    If "Default" is already set in partition_conf, this variable will have no effect.
+    EOD
+  type        = bool
+  default     = false
+}
+
+variable "machine_type" {
+  description = "Compute Platform machine type to use for this partition compute nodes"
+  type        = string
+  default     = "c2-standard-60"
+}
+
+variable "metadata" {
+  type        = map(string)
+  description = "Metadata, provided as a map."
+  default     = {}
+}
+
+variable "node_count_static" {
+  description = "Number of nodes to be statically created"
+  type        = number
+  default     = 0
+}
+
+variable "node_conf" {
+  description = "Map of Slurm node line configuration"
+  type        = map(any)
+  default     = {}
+}
+
+variable "node_count_dynamic_max" {
+  description = "Maximum number of nodes allowed in this partition"
+  type        = number
+  default     = 10
+}
+
+variable "source_image_project" {
+  type        = string
+  description = "Project where the source image comes from. If it is not provided, the provider project is used."
+  default     = ""
+}
+
+variable "source_image_family" {
+  type        = string
+  description = "Source image family."
+  default     = ""
+}
+
+variable "source_image" {
+  type        = string
+  description = "Source disk image."
+  default     = ""
```

Comment:
I am confused by the image variable defaults. I would have expected to see this pointing to the slurm image by default.

Reply:
It defaults to the schedmd-public-images project and the default family name for the hpc-centos-7 slurm image in the underlying module. I've updated the description to make this more clear in a coming commit.